WIP - [WFM Wrapper] Move device specific methods into device lib & invert dependencies between devices and wfm

twoshark opened the pull request.

twoshark committed:
wip have devices declare an use wfm methods

dtp263 commented on lib/Relay/Relay.cpp:
```diff
@@ -0,0 +1,166 @@
+
+
+
+  void Relay::SendHTTP(WiFiClient client)
```
Do you think we should keep this? Weve seen that using a UI kinda wrecks the webserver.

twoshark replied:
Theoretically, that is improved by the division of work across cores. It's nice to have it as a debug option for individual devices if the overhead ends up not being an issue. 

Will test!

dtp263 commented on src/relay/main.cpp:
```diff
@@ -3,19 +3,19 @@
-Relay relay = Relay();
 TaskHandle_t WebServerTaskHandle;
 TaskHandle_t IOTaskHandle;
 WiFiManagerWrapper wfm = WiFiManagerWrapper();
```
since this WiFiManagerWrapper (we could prolly change the name of this at some point soon) still handles the `/specification` endpoint. I think we will want to inject whatever device information we want to include in that spec response. The most basic example could be the device_type. Which in this case would be `relay`?

The benefit of having the wfm do it for all devices will be the standardization of the spec responses.

twoshark replied:
Hmm this is a tricky one, I actually intended to move this path into the Relay class. I just hadn't done it yet.

We can inject the spec in when we instantiate a new WFM in Relay, but I think it should be a shared type that contains maps of paths/parameters/etc. Then wfmw can validate the input and pull data from known keys.

That way the wfm wrapper can stay generic and there's closure around device business logic in the `lib/Relay/`

twoshark committed:
fix: build errors from multiple inputs
- moved shouldSaveConfig into Wfm Wrapper Class
- updated shouldSaveConfig to use an anon func that sets shouldSaveConfig
- delete unused mutex

twoshark committed:
wip
